app/views/editor/level/tasks/TasksTabView.coffee

```diff
@@ -0,0 +1,149 @@
+    'Add programming concepts covered.':'./'
+    'Mark whether it requires a subscription.':'./'
+    'Release to everyone via MailChimp.':'./'
+    'Check completion/engagement/problem analytics.':'./'
+    'Do thorough set decoration.':'./'
+    'Add a walkthrough video.':'./'
+
+  taskMap: ->
+    return @tasks?.map((_obj) -> return (name: _obj.get('name'), complete: (_obj.get('complete') || false)))
+
+  taskArray: ->
+    return @tasks?.toArray()
+
+  findTask: (_name) ->
+    return @tasks.findWhere(name:_name)
+
+  onLoaded: ->
+  onLevelLoaded: (e) ->
+    @level = e.level
+    Task = Backbone.Model.extend({
+      initialize: ->
+        if e?.level?._revertAttributes?.tasks?
+          if _.find(e.level._revertAttributes.tasks, {name:arguments[0].name})
+            @set 'revert', _.find(e.level._revertAttributes.tasks, {name:arguments[0].name})
+          else
+            @set 'revert', arguments[0]
+        else
+          @set 'revert', arguments[0]
+    })
+    TaskList = Backbone.Collection.extend({
+      model: Task
+    })
+    @tasks = new TaskList(@level.get 'tasks')
+    @render()
+
+
+  pushTasks: ->
+    @level.set 'tasks', @taskMap()
+
+  onClickTaskRow: (e) ->
+    if not $(e.target).is('input') and not $(e.target).is('a') and not $(e.target).hasClass('startEdit')
+      task = @findTask($(e.target).closest('tr').data('task').name)
+      checkbox = $(e.currentTarget).find('.taskInput')[0]
+      if task.get 'complete'
+        task.set 'complete', false
+      else
+        task.set 'complete', true
+      checkbox.checked = task.get 'complete'
+      @pushTasks()
+
+  onClickTaskInput: (e) ->
+    task = @findTask($(e.target).closest('tr').data('task').name)
+    task.set 'complete', e.currentTarget.checked
+    @pushTasks()
+
+  onClickStartEdit: (e) ->
+    if $('#curEdit').length is 0
+      task = @findTask($(e.target).closest('tr').data('task').name)
+      task.set 'curEdit', true
+      @render()
+    editDiv = $('#curEdit')[0]
+    editDiv.focus()
+    len = editDiv.value.length * 2
+    editDiv.setSelectionRange len, len
+
+  onKeyDownCurEdit: (e) ->
+    if e.keyCode is 13
+      editDiv = $('#curEdit')[0]
+      task = @findTask($(e.target).closest('tr').data('task').name)
+      potentialTask = @findTask(editDiv.value)
+      if potentialTask and potentialTask isnt task
+        noty
+          timeout: 5000
+          text: 'Task with name already exists!'
+          type: 'error'
+          layout: 'topCenter'
+      else if editDiv.value is ''
+        @tasks.remove task
+        @pushTasks()
+        @render()
+      else
+        task.set 'name', editDiv.value
+        task.set 'curEdit', false
+        @pushTasks()
+        @render()
+
+  onClickCreateTask: (e) ->
+    if $('#curEdit').length is 0
+      @tasks.add
+        name: ''
+        complete: false
+        curEdit: true
+        revert:
+          name: 'null'
+          complete: false
+      @render()
+    editDiv = $('#curEdit')[0]
```

Comment:
These four lines are used in 110-113 as well. Probably should be refactored to an aptly named function, like focusInput. Also, editDiv is always an input element, not a div element, isn't it?